lib/mayaUsd/ufe/UsdUndoMaterialCommands.cpp

```diff
@@ -34,6 +35,45 @@ PXR_NAMESPACE_USING_DIRECTIVE
 namespace MAYAUSD_NS_DEF {
 namespace ufe {
 
+namespace {
+// We will not use the value of UsdUtilsGetMaterialsScopeName() for the material scope.
+static const std::string kDefaultMaterialScopeName("mtl");
+
+#if (UFE_PREVIEW_VERSION_NUM >= 4010)
+bool connectShaderToMaterial(
+    Ufe::SceneItem::Ptr shaderItem,
+    UsdPrim             materialPrim,
+    const std::string&  nodeId)
+{
+    auto                 shaderUsdItem = std::dynamic_pointer_cast<UsdSceneItem>(shaderItem);
```

Comment:
Are we always guaranteed that these casts will work? This being in a local function, I suppose we are guaranteed that the casts will succeed by the caller?

Reply:
It is a good thing that we do not do coverage testing, as this will have me add three unreachable lines of code.

Reply:
It is possible to work around these kinds of issues by moving the function into a utility file and be able to test the function independently from its caller. We'll cross that bridge when we've actually built it. :)